### ISSUE TITLE
releng/docker: minor-release tags

### ISSUE THREAD

**Issue by srenatus** [open-policy-agent/opa#5217]
It would be nice of users were able to use a tag like `openpolicyagent/opa:0.44` and have it resolve to whatever patch release in the 0.44.x series we have out there.

This is similar to how others work, like, golang:1.19 will get you 1.19.2 today.

I think it would require the release process to
1. tag a major release `x.y.0` with `x.y`
2. re-tag `x.y.1` with `x.y` when it's released, and same for any later `x.y.2`, `x.y.3`, ...

**Comment by anderseknert** [comment]
Interesting! I'm curious to learn how this works with image scanners, various "pull policies" as found in Kube, etc. Is there any example of a "runtime" project doing this we could learn from?

**Comment by srenatus** [comment]
It's a feature of convenience. The current behaviour would stay the same, who wants to pin a version can still do that by using 0.46.0 instead of 0.46

**Comment by SVilgelm** [comment]
but the 0.46.0 will not contain any bug fixes from 0.46.1

**Comment by srenatus** [comment]
Exactly. That is just like it is now.

**Comment by SVilgelm** [comment]
yeah, that's why we need something like 0.46 that will be re-tagged with every patch release

**Comment by anderseknert** [comment]
> It's a feature of convenience. The current behaviour would stay the same, who wants to pin a version can still do that by using 0.46.0 instead of 0.46

@srenatus what I meant was, it does not sound convenient to me if I manage 500 OPAs in a cluster, all tagged "v0.100", and I have no idea whether they're _really_ running v0.100.**0** which sports a stop-the-world vulnerability, or the patched v0.100.**1**. It would all depend on when the container started and what version was v0.100 at that point in time. Maybe it's a non-issue, it's not often I have to deal with this sort of stuff :)

**Comment by srenatus** [comment]
@SVilgelm do you know of other examples? ☝️

**Comment by SVilgelm** [comment]
I have a very simple scenario: running some tests for last 3 versions. So in my case I want to always run for the latest patch versions, without changing the test manifests. As a workaround I can create a special scrip, that will ask GitHub or docker for the list of releases, and just pick and use the latest patches for given minors. :) But I would prefer the 0.46 way :)

**Comment by anderseknert** [comment]
Thanks! That sort of confirms my suspicion of this being more of a "compile time" nicety and not something to use at "runtime". Nothing wrong with that of course! Might be good to point out in the docs or something though.
